Tests/Leb128.lean

```diff
@@ -0,0 +1,138 @@
+  let actual := npad7 lebx
+  test "Number 624485, 7-bit padded should be [0,1,0,0,1,1,0,0,0,0,1,1,1,0,1,1,0,0,1,0,1]" $
+    expected == actual &&
+    actual == nplebx &&
+    actual == plebx
+
+def testUnsignedLeb128 : TestSeq := Id.run $ do
+  let expected : ByteArray := ByteArray.mk #[229, 142, 38]
+  let actual := uLeb128 lebx
+  test "Unsigned LEB128 of 624485 should be ByteArray.mk #[ 229, 142, 38 ] and 1 is ByteArray.mk #[1]" $
+    expected.data == actual.data &&
+    (uLeb128 1).data == (ByteArray.mk #[1]).data
+
+def bigN := 1499559017
+
+def testBigNIs233_232_133_203_5 : TestSeq := Id.run $ do
+  let expected : ByteArray := ByteArray.mk #[233, 232, 133, 203, 5]
+  let actual := uLeb128 bigN
+  test "Big endian 1499559017 should be ByteArray.mk #[233, 232, 133, 203, 5]" $
+    expected.data == actual.data
+
+-- 0000010111001011100001011110100011101001
+def testBigNReassembledAfterNPad7Is0000010111001011100001011110100011101001 : TestSeq := Id.run $ do
+  -- 0000010111001011100001011110100011101001 with spaces between each bit:
+  --                          0 0 0 0  0 1 0 1  1 1 0 0  1 0 1 1  1 0 0 0  0 1 0 1  1 1 1 0  1 0 0 0  1 1 1 0  1 0 0 1
+  let expected : List Bit := [0,0,0,0, 0,1,0,1, 1,1,0,0, 1,0,1,1, 1,0,0,0, 0,1,0,1, 1,1,1,0, 1,0,0,0, 1,1,1,0, 1,0,0,1]
+  --                         [0,0,0,0, 0,1,0,1, 1,1,0,0, 1,0,1,1, 1,0,0,0, 0,1,0,1, 1,1,1,0, 1,0,0,0, 1,1,1,0, 1,0,0,1]
+  let actual := reassemble $ npad7 bigN
+  test "Big endian 1499559017, 7-bit padded, reassembled should be [0,0,0,0,0,1,0,1,1,1,0,0,1,0,1,1,1,0,0,0,0,1,0,1,1,1,1,0,1,0,0,0,1,1,0,1,0,0,0,1]" $
+    expected == actual
+
+def signed_lebxs := [ (-123456, ByteArray.mk #[192, 187, 120]), (624485, ByteArray.mk #[229, 142, 38]) ]
+
+def testSignedLeb128 : TestSeq := Id.run $ do
+  let expected : List (Int × ByteArray) := signed_lebxs
+  let actual := List.map (λ (n : Int × ByteArray) => (n.1, sLeb128 n.1)) signed_lebxs
+  test "Signed LEB128 of -123456 should be ByteArray.mk #[192, 187, 120] and 624485 is ByteArray.mk #[229, 142, 38]" $
+    -- Pairwise compair each element's first element and second element between expected and actual.
+    -- Implementation follows:
+    actual.all $
+      fun (n : Int × ByteArray) =>
+        expected.any (fun (m : Int × ByteArray) => n.1 == m.1 && n.2.data == m.2.data)
+
+def testSignedLeb128HitherThither : TestSeq := Id.run $ do
+  -- IO.println s!"{uLeb128 9000} should be 168, 70 (checked with https://npm.runkit.com/%40webassemblyjs%2Fleb128)"
+  test "Signed LEB128 of -1 is [127], signed leb of 9000 is [168, 198, 0], whereas unsigned leb is [168, 70]. And signed leb of 8787 is [211, 196, 0]" $
+    (sLeb128 (-1)).data == (ByteArray.mk #[127]).data &&
+    (sLeb128 9000).data == (ByteArray.mk #[168, 198, 0]).data &&
+    (sLeb128 8787).data == (ByteArray.mk #[211, 196, 0]).data &&
+    -- U is different!
+    (uLeb128 9000).data == (ByteArray.mk #[168, 70]).data
```

Comment:
It's better to do four different tests instead of one with `&&`, for better error reporting. Chain them like `test "test1" (2=2) $ test "test2" (3=3)`